File: src/main/java/org/simbrain/network/gui/actions/dl4j/AddNeuronArrayAction.java
package org.simbrain.network.gui.actions.dl4j;

import org.simbrain.network.gui.NetworkPanel;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class AddNeuronArrayAction extends AbstractAction {


    /**
     * Network panel.
     */
    private final NetworkPanel networkPanel;

    /**
     * Create a new neuron array with the specified network panel.
     *
     * @param networkPanel network panel, must not be null
     */
    public AddNeuronArrayAction(final NetworkPanel networkPanel) {
        super("Add Neuron Array...");

        if (networkPanel == null) {
            throw new IllegalArgumentException("networkPanel must not be null");
        }

        this.networkPanel = networkPanel;
        putValue(SHORT_DESCRIPTION, "Add a neuron array to the network");

    }

    @Override
    public void actionPerformed(final ActionEvent event) {
        networkPanel.showNeuronArrayCreationDialog();
    }

}

File: src/main/kotlin/org/simbrain/network/gui/actions/dl4j/AddNeuronArrayAction.kt
package org.simbrain.network.gui.actions.dl4j

import org.simbrain.network.gui.NetworkPanel
import org.simbrain.network.gui.showNeuronArrayCreationDialog
import java.awt.event.ActionEvent
import javax.swing.AbstractAction
import javax.swing.Action

class AddNeuronArrayAction(val networkPanel: NetworkPanel) : AbstractAction("Add Neuron Array...") {

    override fun actionPerformed(event: ActionEvent) {
        networkPanel.showNeuronArrayCreationDialog()
    }

    init {
        putValue(Action.SHORT_DESCRIPTION, "Add a neuron array to the network")
    }
}